src/univariates.jl

```diff
@@ -306,17 +306,24 @@ cf(d::UnivariateDistribution, t)
 # pdf
 
 """
-    pdf(d::UnivariateDistribution, x::Real)
+    pdf(d::UnivariateDistribution, x::T) where {T<:Real}
 
-Evaluate the probability density (mass) at `x`.
+Evaluate the probability density (mass) at `x`. If `T` is not an `Integer`
+type but `x` is integer, the value is converted to `Int`.
 
 See also: [`logpdf`](@ref).
 """
-pdf(d::UnivariateDistribution, x::Real)
-pdf(d::DiscreteUnivariateDistribution, x::Int) = throw(MethodError(pdf, (d, x)))
-pdf(d::DiscreteUnivariateDistribution, x::Integer) = pdf(d, round(Int, x))
 pdf(d::DiscreteUnivariateDistribution, x::Real) = isinteger(x) ? pdf(d, round(Int, x)) : 0.0
 
+"""
```

Comment:
This docstring would better be merged with the one above IMHO.

Reply:
done thanks